packages/web-components/src/rating-display/rating-display.ts

```diff
@@ -0,0 +1,122 @@
+import { attr, FASTElement, nullableNumberConverter } from '@microsoft/fast-element';
+import { RatingDisplayColor, RatingDisplaySize } from './rating-display.options.js';
+
+/**
+ * The base class used for constructing a fluent-rating-display custom element
+ * @public
+ */
+export class RatingDisplay extends FASTElement {
+  /**
+   * The internal {@link https://developer.mozilla.org/docs/Web/API/ElementInternals | `ElementInternals`} instance for the component.
+   *
+   * @internal
+   */
+  public elementInternals: ElementInternals = this.attachInternals();
+
+  /**
+   * The color of the rating display icons.
+   *
+   * @public
+   * @default `marigold`
+   * @remarks
+   * HTML Attribute: `color`
+   */
+  @attr
+  public color: RatingDisplayColor = RatingDisplayColor.marigold;
+
+  /**
+   * Renders a single filled icon with a label next to it.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `compact`
+   */
+  @attr({ attribute: 'compact', mode: 'boolean' })
+  public compact: boolean = false;
+  protected compactChanged(): void {
+    if (this.compact) {
+      this.max = 1;
+    }
+  }
+
+  /**
+   * The number of ratings.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `count`
+   */
+  @attr({ converter: nullableNumberConverter })
+  public count?: number;
+
+  /**
+   * The maximum possible value of the rating.
+   * This attribute determines the number of icons displayed.
+   * Must be a whole number greater than 1.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `max`
+   */
+  @attr({ converter: nullableNumberConverter })
+  public max: number = 5;
+
+  /**
+   * The size of the component.
+   *
+   * @public
+   * @default 'medium'
+   * @remarks
+   * HTML Attribute: `size`
+   */
+  @attr
+  public size: RatingDisplaySize = RatingDisplaySize.medium;
+
+  /**
+   * The value of the rating.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `value`
+   */
+  @attr({ converter: nullableNumberConverter })
+  public value: number = 0;
+
+  private intlNumberFormatter = new Intl.NumberFormat();
+
+  public constructor() {
+    super();
+
+    this.elementInternals.role = 'img';
```

Comment:
Can this just be set on the `connectedCallback`?

Reply:
Since this doesn't every change based on attributes, I think the constructor is (for now) the appropriate place.

Reply:
@chrisdholt I was wondering about this from other PRs, since we didn't historically use the constructor in component classes, Thanks for the explanation!